content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1441 @@
+                <button type="reset">Reset</button>
+                <button type="submit">Confirm</button>
+            </div>
+        </form>
+    </dialog>
+
+</body>
+```
+
+6. To incorporate the refactored markup, handle the HTML dialog element and well as the new account data section replace the contents of `view/renderer.js` with the following code:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/3_renderer.js", language="js") }}
+
+Now you need an XRPL account address to monitor. If you already have one or know where to find an example, you can now run the application by executing:
+
+```console
+npm run start
+```
+If you are new to the XRPL and need an account address, [you can get accounts on the testnet](https://learn.xrpl.org/course/code-with-the-xrpl/lesson/create-accounts-and-send-xrp/). There you can also use the sandbox to issue XRP transactions, which then should show up in our app.
+
+On, startup, the application should display a simple dialog prompting the user for an XRPL account address. After entering the address the application should display some basic information about that account and about the ledger.
+
+To run the reference application found in `content/_code-samples/build-a-wallet/desktop-js` for this step, run:
+
+```console
+npm run account
+```
+
+### 4. Show Account's Transactions
+
+**Full code for this step:** 
+[`4_tx-history.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/4_tx-history.js).
+[`library/3_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/3_helper.js).
+[`library/4_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/4_helper.js).
+[`view/4_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/4_tx-preload.js).
+[`view/4_tx-history.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/4_tx-history.html).
+[`view/4_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/4_tx-renderer.js).
+
+At this point, our wallet shows the account's balance getting updated, but doesn't give us any clue about how this state came about, namely the actual transactions that caused the updates. So, our next step is to display the account's up to date transaction history using subscriptions once again:
+
+![Screenshot: Step 4, show transaction history](img/javascript-wallet-4.png)
+
+1. In the `library` folder, add a new file `4_helpers.js`. Then add the following helper function to that file:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/library/4_helpers.js", language="js") }}
+
+2. Now, in `index.html`, require the new helper function at the bottom of the import section like so:
```

Comment:
There is no file called `index.html` currently - did you mean `index.js`?

Reply:
Done, should have read my Kahnemann on that one :)